scripts/yum-plugins/ptoken.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/python
+
+# Drop this file into /usr/lib/yum-plugins/
+# Enable it by creating conf file /etc/yum/pluginconf.d/ptoken.conf:
+#   [main]
+#   enabled=1
+#
+# Configure it by:
+# yum-config-manager --setopt=<repo-name>.ptoken=file://<token-file-path> --save
+
+from yum import config
+from yum.plugins import TYPE_CORE
+
+requires_api_version = '2.5'
+plugin_type = (TYPE_CORE,)
+
+def config_hook(conduit):
+    config.RepoConf.ptoken = config.BoolOption(False)
+
+def init_hook(conduit):
+    with open('/etc/xensource/ptoken') as f:
+        ptoken = f.read().strip()
+
+    repos = conduit.getRepos()
+    for name in repos.repos:
+        repo = repos.repos[name]
+        if repo.getConfigOption('ptoken'):
+            repo.http_headers['cookie'] = "pool_secret=" + ptoken
```

Comment:
If we're sending the pool token along with every request we should probably forbid http:// URIs in our yum config where ptoken is used. i.e. you either use an internal mirror through http, or the one through ptoken but that must be https.

Reply:
Yes, that is already the case. See:
* https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/repository.ml#L466
* https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/fileserver.ml#L83